Assets/Minigames/Sorting/Scripts/Sound/MusicController.cs

```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class MusicController : MonoBehaviour
+{
+    public GameObject audioManager;
+    public Sprite offSprite;
+    public Sprite onSprite;
+
+    private Image buttonImage;
+    private bool isPlaying = true;
+
+    public void ToggleMusic()
+    {
+        buttonImage = GetComponent<Image>();
+        AudioSource backgroundMusic = audioManager.GetComponentInChildren<AudioSource>();
```

Comment:
If I remember correctly, the audiomanager object has multiple child audio sources. How can you tell that the one you got is the one responsible for the background music?
Also, there's a SoundManager script/object in the scene that you should use to add the muting functionality to, not here. This way the functionality will be used for the entire game and not just one minigame - keeps things consistnet

Reply:
Hi Shen, I have some confusion here.

1. Is the old 'Data Charger.unity' I was working on before the current 'Data Charger Standalone.unity'? Because before there was only one scene and in that scene there was 'Camera' and 'Audio Manager (with a script SoundManager)', now the current 'Data Charger.unity' doesn't have them.

2. There is only 1 audio source under 'AudioManager' Object for background music, and the other sound effects are in Charge Station QN/QL. (Only in the old 'Data Charger.unity' where I worked before and now in the current 'Data Charger Standalone.unity')

3. I was testing the current 'Data Charger Standalone.unity' and found that some of the original basic functions are not working, but I don't know why.

4. I'll rework the code for controlling the BGM and add it to the SoundManager script, but I'm not sure how to control the BGM in the current ‘Data Charger.unity’ yet, as it doesn't have the 'Audio Manager Object' in it as well as the 'Sound Manager Script'.

Reply:
Just a side note, if this isn't an easy fix, I can redo all my history work on the correct version.

Reply:
Hey Zike,

1. The standalone has them because that's the minigame on its own without the full game, so that you can play it in the minigames section of the [webpage ](https://dlg.tlu.ee/methodyca/mini-games/data-charger).
In the full game, the Sound Manager isn't in that scene because the Data Charger scene is loaded on top of the game world scene. And the Sound Manager is in that world scene already.

2. If you look at SoundManager script and in the PlaySFX function, every time a new SFX is played a new AudioSource component is added to Sound Manager game object. `GetComponentInChildren()` will in practice look at the game object itself first before its child objects and thus will return to you the SFX components - especially so if there's a looping sfx. Check out the `Sound` script as well in the same folder of the SoundManager script.

3. I don't know what you mean by "original basic functions" you have to clarify and better on discord chat.

4. Refer to point 1

You don't have to redo anything. Just modify and update the current situation. The sorting minigame branch has also been updated now to match with Development branch